File: LeetCode/Study Plans/Binary Search I/Day 6/1539. Kth Missing Positive Number.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var findKthPositive = function(arr, k) {
    let ans = 1, i = 0;
    while(i < arr.length && k > 0) {
        if(arr[i] == ans) i++;
        else k--;
        ans++;
    }
    while(k--) ans++;
    return --ans;
};

File: LeetCode/Study Plans/Binary Search I/Day 6/1539. Kth Missing Positive Number.cpp
// Question 2

// Problem Link
// https://leetcode.com/problems/kth-missing-positive-number/

// Solution:
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        int i = 1;
        while(k > 0) {
            if(find(arr.begin(), arr.end(), i) == arr.end()) k--;
            i++;
        }
        return --i;
    }
};